mne_bids/tests/test_write.py

```diff
@@ -1395,3 +1396,139 @@ def test_write_does_not_alter_events_inplace():
+
+    # Set descriptions: string.
+    bads = 'MEG 0112'
+    descriptions = 'Really bad!'
+    mark_bad_channels(channels=bads, descriptions=descriptions,
+                      bids_basename=bids_basename, bids_root=bids_root,
+                      kind='meg')
+    _bids_validate(bids_root)
+    tsv_data = _from_tsv(channels_fname)
+    assert 'Really bad!' in tsv_data['status_description']
+
+    # Test that we raise if number of channels doesn't match number of
+    # descriptions.
+    bads = ['MEG 0112', 'MEG 0131']
+    descriptions = ['Really bad!']
+    with pytest.raises(ValueError, match='must match'):
+        mark_bad_channels(channels=bads, descriptions=descriptions,
+                          bids_basename=bids_basename, bids_root=bids_root,
+                          kind='meg')
+
+    # Test that we create missing columns.
+    tsv_data = _from_tsv(channels_fname)
+    del tsv_data['status'], tsv_data['status_description']
+    _to_tsv(tsv_data, channels_fname)
+
+    bads = 'MEG 0112'
+    descriptions = 'Really bad!'
+    mark_bad_channels(channels=bads, descriptions=descriptions,
+                      bids_basename=bids_basename, bids_root=bids_root,
+                      kind='meg')
+    _bids_validate(bids_root)
+    tsv_data = _from_tsv(channels_fname)
+    assert 'status' in tsv_data
+    assert 'status_description' in tsv_data
+
+    # Test that we leave existing entries unmodified by default.
+    old_bads = ['MEG 0112', 'MEG 0131']
+    old_descriptions = ['Really bad!', 'Even worse.']
+    new_bads = ['EEG 053']
+    new_descriptions = ['Just testing']
+
+    mark_bad_channels(channels=old_bads, descriptions=old_descriptions,
+                      bids_basename=bids_basename, bids_root=bids_root,
+                      kind='meg')
+    mark_bad_channels(channels=new_bads, descriptions=new_descriptions,
+                      bids_basename=bids_basename, bids_root=bids_root,
+                      kind='meg')
+    _bids_validate(bids_root)
+
+    raw = read_raw_bids(bids_basename=bids_basename, bids_root=bids_root,
+                        kind='meg', verbose=False)
+    assert set(raw.info['bads']) == set(old_bads + new_bads)
+
+    tsv_data = _from_tsv(channels_fname)
+    assert all([d in tsv_data['status_description']
+                for d in old_descriptions + new_descriptions])
+
+    # Test that existing entries are discarded if `overwrite=True`.
+    old_bads = ['MEG 0112', 'MEG 0131']
+    old_descriptions = ['Really bad!', 'Even worse.']
+    new_bads = ['EEG 053']
+    new_descriptions = ['Just testing']
+
+    mark_bad_channels(channels=old_bads, descriptions=old_descriptions,
+                      bids_basename=bids_basename, bids_root=bids_root,
+                      kind='meg')
+    mark_bad_channels(channels=new_bads, descriptions=new_descriptions,
+                      bids_basename=bids_basename, bids_root=bids_root,
+                      kind='meg', overwrite=True)
```

Comment:
do you have tests for kind=None ? just one would be enough

Reply:
Added in 65fa255